packages/gatsby-transformer-asciidoc/src/gatsby-node.js

```diff
@@ -1,34 +1,54 @@
+"use strict"
+
 const asciidoc = require(`asciidoctor.js`)()
 
 async function onCreateNode(
   {
     node,
     actions,
+    pathPrefix,
     loadNodeContent,
     createNodeId,
     reporter,
     createContentDigest,
   },
   pluginOptions
 ) {
-  // Filter out non-adoc content
-  if (!node.extension || node.extension !== `adoc`) {
+  var extensionsConfig = pluginOptions.fileExtensions
+
+  // make extensions configurable and use adoc and asciidoc as default
+  var supportedExtensions =
+    typeof extensionsConfig != `undefined` && extensionsConfig instanceof Array
+      ? extensionsConfig
+      : [`adoc`, `.asciidoc`]
+  var currentExtension = node.extension
+  var isSupportedExtension = supportedExtensions.indexOf(currentExtension) > -1
+
+  if (!isSupportedExtension) {
     return
   }
 
-  const { createNode, createParentChildLink } = actions
-  // Load Asciidoc contents
-  const content = await loadNodeContent(node)
-  // Load Asciidoc file for extracting
+  // changes the incoming imagesdir option to take the
+  var changedImagesDir = resolveImagesDir(
+    pathPrefix,
+    pluginOptions.attributes.imagesdir
```

Comment:
`pluginOptions.attributes` will not always be defined, so we need to do some sanity checks to avoid "can't read imagesdir of undefined" errors

Reply:
good point